packages/react-components/src/components/MessageThread.tsx

```diff
@@ -728,10 +728,10 @@ export const MessageThreadWrapper = (props: MessageThreadProps): JSX.Element =>
         return;
       }
       try {
-        await onDeleteMessage(messageId);
         // reset deleted message label in case if there was a value already (messages are deleted 1 after another)
         setDeletedMessageAriaLabel(undefined);
         setLatestDeletedMessageId(messageId);
+        await onDeleteMessage(messageId);
```

Comment:
Should we leave a comment here explaining this line need to be below the 2 lines above because of this race condition?

Reply:
Good catch, added!